assets/src/design-system/components/keyboard/index.js

```diff
@@ -14,5 +14,390 @@
  * limitations under the License.
  */
 
-export { isPlatformMacOS } from './isPlatformMacOS';
-export { prettifyShortcut } from './prettifyShortcut';
+/**
+ * External dependencies
+ */
+import Mousetrap from 'mousetrap';
+import { useEffect, createRef, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import useBatchingCallback from '../../utils/useBatchingCallback';
+import { useContext } from '../../utils/context';
+import Context from './context';
+
+const PROP = '__WEB_STORIES_MT__';
+const NON_EDITABLE_INPUT_TYPES = [
+  'submit',
+  'button',
+  'checkbox',
+  'radio',
+  'image',
+  'file',
+  'range',
+  'reset',
+  'hidden',
+];
+const CLICKABLE_INPUT_TYPES = [
+  'submit',
+  'button',
+  'checkbox',
+  'radio',
+  'image',
+  'file',
+  'reset',
+];
+
+const globalRef = createRef();
+
+function setGlobalRef() {
+  if (!globalRef.current) {
+    globalRef.current = document.documentElement;
+  }
+}
+
+/**
+ * See https://craig.is/killing/mice#keys for the supported key codes.
+ *
+ * @param {Node|{current: Node}} refOrNode Node or reference to one.
+ * @param {string|Array|Object} keyNameOrSpec Single key name or key spec.
+ * @param {string|undefined} type Event type, either 'keydown', 'keyup', or undefined to automatically determine it.
+ * @param {function(KeyboardEvent)} callback Callback.
+ * @param {Array|undefined} deps The effect's dependencies.
+ */
+function useKeyEffectInternal(
+  refOrNode,
+  keyNameOrSpec,
+  type,
+  callback,
+  deps = undefined
+) {
+  const { keys } = useContext(Context);
+  //eslint-disable-next-line react-hooks/exhaustive-deps
+  const batchingCallback = useBatchingCallback(callback, deps || []);
+  useEffect(
+    () => {
+      const node =
+        typeof refOrNode.current !== 'undefined'
+          ? refOrNode.current
+          : refOrNode;
+      if (!node) {
+        return undefined;
+      }
+      if (
+        node.nodeType !== /* ELEMENT */ 1 &&
+        node.nodeType !== /* DOCUMENT */ 9
+      ) {
+        throw new Error('only an element or a document node can be used');
```

Comment:
So throwing Errors should really be reserved for when we either have a error boundary (in the UI) or there is no way to recover. I think we should return `null` or `undefined` here instead. I hesitate to use `Error` objects for implementation issues. Not runtime. I am open to ideas around this.